python/taichi/lang/matrix.py

```diff
@@ -185,6 +185,11 @@ def linearize_entry_id(self, *args):
             args = args + (0, )
         assert 0 <= args[0] < self.n
         assert 0 <= args[1] < self.m
+        # TODO(#1004): See if it's possible to support indexing at runtime
+        for i, a in enumerate(args):
+            assert isinstance(
+                a, int
+            ), f'The {i}-th index of a Matrix/Vector must be a compile-time constant integer, got {a}'
```

Comment:
```suggestion
            ), f'The {i+1}-th index of a Matrix/Vector must be a compile-time constant integer, got {a}'
```
Btw, does 0-th make sense in English? Like ground floor in UK vs first floor in US?

Reply:
I prefer `The {i}-th index`. "Zeroth" makes sense, especially to programmers.